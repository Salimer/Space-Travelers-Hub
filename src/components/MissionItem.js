import Badge from 'react-bootstrap/Badge';
import Button from 'react-bootstrap/Button';
import PropTypes from 'prop-types';

const MissionItem = (props) => {
  const {
    id, name, description, reserved, handleClick,
  } = props;

  return (
    <tr>
      <td style={{ fontWeight: 'bold' }}>{name}</td>
      <td>{description}</td>
      <td className="align-middle" style={{ textAlign: 'center' }}>
        {reserved ? <Badge bg="success">Active Member</Badge>
          : <Badge bg="secondary">NOT A MEMBER</Badge>}
      </td>
      <td className="align-middle" style={{ width: '11rem', textAlign: 'center' }}>
        <Button
          data-testid="mission-item-button"
          variant={reserved ? 'outline-danger' : 'outline-dark'}
          type="button"
          onClick={(e) => {
            handleClick(e, { id, reserved });
          }}
        >
          {reserved ? 'Leave mission' : 'Join mission'}
        </Button>
      </td>
    </tr>
  );
};

MissionItem.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  reserved: PropTypes.bool.isRequired,
  handleClick: PropTypes.func.isRequired,
};

export default MissionItem;
