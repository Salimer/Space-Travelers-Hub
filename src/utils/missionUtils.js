const filterReserved = (missions) => missions.filter((mission) => mission.reserved);

export default filterReserved;
