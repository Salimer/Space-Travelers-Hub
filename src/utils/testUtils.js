import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';

const withRouter = (component) => (
  <BrowserRouter>
    {component}
  </BrowserRouter>
);

const withProvider = (component) => (
  <Provider store={store}>
    {component}
  </Provider>
);

export { withProvider, withRouter };
