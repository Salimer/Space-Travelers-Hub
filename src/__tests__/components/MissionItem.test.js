import renderer from 'react-test-renderer';
import MissionItem from '../../components/MissionItem';
import { withProvider } from '../../utils/testUtils';

describe('testing MissionItem component', () => {
  // Item is just for rendering, user interactions are tested on the mission lists
  test('render', () => {
    const tree = renderer.create(withProvider(<MissionItem id="SH44" name="Misison1" description="Mock description" reserved={false} handleClick={() => {}} />)).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
