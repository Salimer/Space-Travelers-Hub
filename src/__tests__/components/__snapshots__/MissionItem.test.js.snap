// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing MissionItem component render 1`] = `
<tr>
  <td
    style={
      Object {
        "fontWeight": "bold",
      }
    }
  >
    Misison1
  </td>
  <td>
    Mock description
  </td>
  <td
    className="align-middle"
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <span
      className="badge bg-secondary"
    >
      NOT A MEMBER
    </span>
  </td>
  <td
    className="align-middle"
    style={
      Object {
        "textAlign": "center",
        "width": "11rem",
      }
    }
  >
    <button
      className="btn btn-outline-dark"
      data-testid="mission-item-button"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Join mission
    </button>
  </td>
</tr>
`;
